### DISCUSSION TITLE
How to get the git-sync password/token from Vault?

### DISCUSSION THREAD

**Discussion by gowdra01** [airflow-helm/charts#568] [Questions & Answers] [answered]
### Checks

- [X] I have checked for [existing issues](https://github.com/airflow-helm/charts/issues).
- [X] I have looked through the [FAQ](https://github.com/airflow-helm/charts/tree/main/charts/airflow#faq---airflow).

### Question

I would like to know how do we pass the TOKEN from the vault
```
    ##   url: "https://USERNAME:TOKEN@github.com/USERNAME/REPOSITORY.git"
```
this is the git url and I need to pass this token from the vault .. how do we do it in the value file?

**Comment by thesuperzapper** [comment, marked as answer]
@gowdra01 You can set the git-sync HTTP username/token from a Kubernetes Secret with `dags.gitSync.httpSecret`, `dags.gitSync.httpSecretUsernameKey`, and `dags.gitSync.httpSecretPasswordKey`. There are a number of ways to automatically create Kubernetes Secrets from your Vault, for example:
- [secrets-store-csi-driver](https://secrets-store-csi-driver.sigs.k8s.io/topics/sync-as-kubernetes-secret.html)
- [external-secrets](https://github.com/external-secrets/external-secrets)
